### PR TITLE
Ovirt module refactor

### PR DESCRIPTION
**Pull request by joshainglis**
Finally got time to circle back on this.
Have rebased onto the latest ansible-modules-extra devel branch, and squashed the commits from myself, @firstyear and @vincentvdk and I believe should be ready for final merging.

This pulls together all commits from and supersedes the following PRs: #583 #589 #707 #925 

cc: @robynbergeron @gregdek

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @joshainglis. @vincentvdk please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by gregdek** [comment]
(Also, @vincentvdk, if you want to add @joshainglis as a maintainer going forward, lemme know, I can do that)

**Comment by sivel** [comment]
```
============================================================================
./cloud/misc/ovirt.py
============================================================================
ERROR: version_added for new option (authorized_key_user) should be 2.1. Currently 2.0
ERROR: version_added for new option (tag) should be 2.1. Currently 2.0
ERROR: version_added for new option (static_ip_address) should be 2.1. Currently 2.0
ERROR: version_added for new option (wait_for_state) should be 2.1. Currently 2.0
ERROR: version_added for new option (static_gateway) should be 2.1. Currently 2.0
ERROR: version_added for new option (instance_host_name) should be 2.1. Currently 2.0
ERROR: version_added for new option (authorized_key_file) should be 2.1. Currently 2.0
ERROR: version_added for new option (static_netmask) should be 2.1. Currently 2.0
ERROR: version_added for new option (poll_frequency) should be 2.1. Currently 2.0
ERROR: version_added for new option (custom_script) should be 2.1. Currently 2.0
ERROR: version_added for new option (authorized_key_user_groups) should be 2.1. Currently 2.0
ERROR: version_added for new option (cluster) should be 2.1. Currently 0.0
ERROR: version_added for new option (nic_name) should be 2.1. Currently 2.0
ERROR: version_added for new option (poll_timeout) should be 2.1. Currently 2.0
```

**Comment by gregdek** [comment]
@vincentvdk This change is still pending your review; do you have time to take a look and comment? Please comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by gregdek** [comment]
Thanks @joshainglis for this PR. A maintainer of this module has asked for revisions to this PR. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by joshainglis** [comment]
@sivel @gregdek Revision addressing https://github.com/ansible/ansible-modules-extras/pull/1543#issuecomment-187461654 has been added: https://github.com/ansible/ansible-modules-extras/pull/1543/commits/8d04ac1ffecf748ce0021e188465d29acd42b12f

**Comment by gregdek** [comment]
Thanks @joshainglis for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by joshainglis** [comment]
@gregdek rebased onto current devel. ready_for_review

**Comment by gregdek** [comment]
Thanks @joshainglis. To the current maintainers, @vincentvdk please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @joshainglis for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @joshainglis for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@joshainglis A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@joshainglis Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by joshainglis** [comment]
@gregdek "ready_for_review"

**Comment by gregdek** [comment]
Thanks @joshainglis. To the current maintainers, @karmab please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @joshainglis for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by joshainglis** [comment]
@gregdek "ready_for_review"

**Comment by gregdek** [comment]
Thanks @joshainglis. To the current maintainers, @karmab please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by joshainglis** [comment]
Hi @gregdek @robynbergeron, this PR has been open for quite some time. Is there any chance we can get a set of eyes cast over it for review?

**Comment by gregdek** [comment]
@joshainglis how do you feel about becoming a reviewer for this module?

**Comment by joshainglis** [comment]
@gregdek I'd be happy to.

**Comment by gregdek** [comment]
OK @joshainglis -- you are now listed as a maintainer for the module. I'll kick this PR and you should end up with shipit rights; please be sure to follow the review process closely before you give the shipit. Thanks for your work so far, and your patience, and for agreeing to maintain this module. :) (Wait for the bot to run again, sometime in the next 6 hours or so.)

**Comment by joshainglis** [comment]
Thanks @gregdek :)
I'll ensure the review process if followed.

**Comment by gregdek** [comment]
Thanks again to @joshainglis. This module is going into community review.

We notice that you are one of the maintainers (@joshainglis, @karmab) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by karmab** [comment]
my bad, will take a look to....

**Comment by gregdek** [comment]
Thanks again to @joshainglis for this PR, and thanks @joshainglis, @karmab for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by alikins** [comment]
Looks like a spurious 'shipit' via (likely the quoted 'shipit' from https://github.com/ansible/ansible-modules-extras/pull/1543#issuecomment-226128802). Removing 'shipit' label for now.

How backwards compatible is this version of the module with the 2.1 version?

File: cloud/misc/ovirt.py [hunk 1]
```diff
@@ -201,320 +232,934 @@
...
+                required=True
+            ),
+            url=dict(
+                type='str',
+                required=True
+            ),
+            instance_name=dict(
+                type='str',
+                required=True,
+                aliases=['instance_name']
```
Comment (alikins) [inline cloud/misc/ovirt.py:964]: ?  Is the alias here intended to be 'vmname' ?

File: cloud/misc/ovirt.py [hunk 2]
```diff
@@ -201,320 +232,934 @@
...
+                aliases=['zone']
+            ),
+            instance_disksize=dict(
+                type='str',
+                aliases=['vm_disksize']
+            ),
+            instance_cpus=dict(
+                type='str',
+                default=1,
+                aliases=['instance_cpus']
```
Comment (alikins) [inline cloud/misc/ovirt.py:989]: Similarly, should the alias here be 'vmcpus' ?

File: cloud/misc/ovirt.py [hunk 3]
```diff
@@ -201,320 +232,934 @@
...
+            ),
+            instance_nic=dict(
+                type='str',
+                aliases=['instance_nic']
```
Comment (alikins) [inline cloud/misc/ovirt.py:993]: vmnic

File: cloud/misc/ovirt.py [hunk 4]
```diff
@@ -201,320 +232,934 @@
...
+            ),
+            instance_network=dict(
+                type='str',
+                default='rhevm',
+                aliases=['instance_network']
```
Comment (alikins) [inline cloud/misc/ovirt.py:998]: vmnetwork

File: cloud/misc/ovirt.py [hunk 5]
```diff
@@ -201,320 +232,934 @@
...
+            ),
+            instance_mem=dict(
+                type='str',
+                aliases=['instance_mem']
```
Comment (alikins) [inline cloud/misc/ovirt.py:1002]: vmmem

File: cloud/misc/ovirt.py [hunk 6]
```diff
@@ -201,320 +232,934 @@
...
+            ),
+            instance_type=dict(
+                type='str',
+                default='server',
+                aliases=['instance_type'],
```
Comment (alikins) [inline cloud/misc/ovirt.py:1007]: vmtype

File: cloud/misc/ovirt.py [hunk 7]
```diff
@@ -201,320 +232,934 @@
...
+                choices=['thin', 'preallocated']
+            ),
+            disk_int=dict(
+                type='str',
+                default='virtio',
+                choices=['virtio', 'ide']
+            ),
+            instance_os=dict(
+                type='str',
+                aliases=['instance_os']
```
Comment (alikins) [inline cloud/misc/ovirt.py:1022]: vmos

File: cloud/misc/ovirt.py [hunk 8]
```diff
@@ -201,320 +232,934 @@
...
+            ),
+            instance_cores=dict(
+                type='str',
+                default=1,
+                aliases=['instance_cores']
```
Comment (alikins) [inline cloud/misc/ovirt.py:1027]: vmcores

**Comment by gregdek** [comment]
Thanks again to @joshainglis for this PR, and thanks @joshainglis, @karmab for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by alikins** [comment]
Need confirmation about the alias names added here.

**Comment by gregdek** [comment]
@joshainglis A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@joshainglis Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@joshainglis A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@joshainglis one last ping on this before I close... got any time to rebase and revise?

**Comment by gregdek** [comment]
@joshainglis A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@joshainglis Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@joshainglis A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@joshainglis Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Closing due to contributor inactivity.

**Pull request closed without merging**